daemons/pacemakerd/pcmkd_subdaemons.c

```diff
@@ -546,47 +545,41 @@ child_liveness(pcmk_child_t *child)
     const gid_t *ref_gid;
     const char *name = pcmk__server_name(child->server);
     int rc = pcmk_rc_ipc_unresponsive;
+    int legacy_rc = pcmk_ok;
     pid_t ipc_pid = 0;
 
-    if (child->endpoint == NULL
-            && (child->pid <= 0 || child->pid == PCMK__SPECIAL_PID)) {
```

Comment:
We can safely assume it since we're using `enum pcmk_ipc_server` values (other than unknown) and those all return non-NULL. I'll add some notes to the APIs explicitly guaranteeing that.

The pid doesn't matter because of the `&&`. Endpoint is never NULL so PID doesn't matter. The original design intent was to support theoretical subdaemons that didn't provide IPC, which is not worth the added complexity since it's extremely unlikely.

Reply:
> The pid doesn't matter because of the &&

Ah, duh.